mantidproject/mantid
# Retain globals in function browser when a new function is added

**Description of work.**
This PR makes it so the global parameters are carried over when a new function is added to the function browser. Previously, when switching from a single to composite function the browser was cleared (and hence the globals). 

**To test:**
1. Open the Muon Analysis 2 GUI
2. Load in any data (e.g MUSR62260)
3. Go to the fitting tab
4. Change the simultaneous  fit (so global checkbox is shown)
5. Add a function
6. Pick a global
7. Add a new function, globals should be retained
8. Remove that new function, globals should be retained

Fixes #29084 . 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
